daml-foundations/daml-ghc/tests/ConsumingTests.daml

```diff
@@ -0,0 +1,43 @@
+daml 1.2
+module ConsumingTests where
+
+template T
+  with
+    p : Party
+  where
+    signatory p
+
+    nonconsuming choice NoConsume : Int
+      controller p
+      do
+        return 41
+
+    choice PreConsume : Int
```

Comment:
Can we have a PreConsumeBad that has a preconsume and a fetch that does submitFail, so we can differentiate between the two.

Reply:
✔️